lib/azure/armrest/model/base_model.rb

```diff
@@ -62,7 +64,18 @@ def initialize(json)
           @json = json
         end
 
-        @ostruct = OpenStruct.new(hash)
+        ostruct = OpenStruct.new(hash)
+        keys = ostruct.to_h.keys
+
+        # If the method already exists then create an "_alias" method for it.
+        methods.each do |m|
+          if keys.include?(m)
+            ostruct.class.instance_eval{ alias_method "_#{m}", m }
+            instance_eval{ define_singleton_method("_#{m}"){ ostruct[m] } }
```

Comment:
That did not work for me either. I've updated the PR for now to remove the unnecessary instance_eval. We can go from there.